apps/schema2/src/components/SchemaBeta.vue

```diff
@@ -0,0 +1,240 @@
+<template>
+  <div class="container-fluid">
+    <div class="d-flex justify-content-between">
+      <div class="form-inline">
+        <h1>Schema editor: {{ schema.name }}</h1>
+        <span v-if="schema.tables">
+          <ButtonAction @click="saveSchema" class="ml-2">Save</ButtonAction>
+          &nbsp;
+          <ButtonAction @click="loadSchema" class="ml-2">Reset</ButtonAction>
+        </span>
+      </div>
+      <div>
+        <span v-if="schema.tables">
+          <JsonView :value="schema" />
+          <ButtonAction @click="showDiagram = !showDiagram">
+            {{ showDiagram ? "Hide" : "Show" }} Diagram
+          </ButtonAction></span
+        >
+      </div>
+    </div>
+    <MessageError v-if="graphqlError">{{ graphqlError }}</MessageError>
+    <MessageWarning v-if="warning">{{ warning }}</MessageWarning>
+    <MessageSuccess v-if="success">{{ success }}</MessageSuccess>
+    <Spinner v-if="loading === true || !schema.tables" />
+    <div v-else class="row">
+      <div class="col-2 bg-white">
+        <div class="fixedContainer mr-n3 overflow-auto">
+          <SchemaToc v-model="schema" v-if="schema.tables" />
+        </div>
+      </div>
+      <div class="bg-white col ml-2 overflow-auto">
+        <div>
+          <Yuml
+            :schema="schema"
+            :key="JSON.stringify(schema)"
+            v-if="showDiagram"
+          />
+          <SchemaView v-model="schema" v-if="schema.tables" />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<style scoped>
+.fixedContainer {
+  position: -webkit-sticky; /* Safari */
+  position: sticky;
+  max-height: 100vh;
+  top: 0;
+}
+</style>
+
+<script>
+import { request } from "graphql-request";
+import Yuml from "./Yuml.vue";
+import SchemaView from "./SchemaView.vue";
+import SchemaToc from "./SchemaToc.vue";
+import JsonView from "./JsonView.vue";
+import {
+  ButtonAction,
+  MessageError,
+  MessageSuccess,
+  MessageWarning,
+  Spinner,
+} from "molgenis-components";
+
+export default {
+  components: {
+    Yuml,
+    SchemaView,
+    ButtonAction,
+    MessageError,
+    MessageWarning,
+    MessageSuccess,
+    SchemaToc,
+    Spinner,
+    JsonView,
+  },
+  data() {
+    return {
+      schema: {},
+      loading: false,
+      graphqlError: null,
+      warning: null,
+      success: null,
+      showDiagram: false,
+    };
+  },
+  methods: {
+    saveSchema() {
+      this.graphqlError = null;
+      this.warning = "submitting changes";
+      let tables = this.schema.tables;
+      this.schema.tables = null; //hide it all
+
+      //transform subclasses back into their original tables.
+      //create a map of tables
+      let tableMap = tables.reduce((map, table) => {
+        map[table.name] = table;
+        table.subclasses.forEach((subclass) => (map[subclass.name] = subclass));
+        delete table.subclasses;
+        return map;
```

Comment:
could do with test

Reply:
I don't agree because this code is on the 'critical' path so issues will be immediately found when using it. However, if you have specific edge case that need testing that would help me to create value adding tests that would not simply reproduce what users do on every call.  <= while I am writing I feel we would have more befits from an e2e test using some representatively complex data model.

Reply:
wontfix for now.